crates/re_log_types/src/time_real.rs

```diff
@@ -64,16 +67,38 @@ impl From<i64> for TimeReal {
 }
 
 impl From<f32> for TimeReal {
+    /// Saturating cast
     #[inline]
     fn from(value: f32) -> Self {
-        Self(FixedI128::from_num(value))
+        assert!(!value.is_nan());
```

Comment:
Crashing here would probably result in loss of user data; we should to `warn!()` and default to some value instead.